### PR TITLE
Quviq ledger test work

### PR DESCRIPTION
**Pull request by MaximilianAlgehed**
@TimSheard please have a look at this as soon as you can. Note the approach we take to generating valid instances of `NewEpochState` in `LedgerTests.hs`. The neat thing is that most things work "out of the box" here.

However, I'm worried that the implementation of the constraints that I've been poking around with (in `Examples.hs`) are not the same instances that are used in the staged approach that you were working on. Let's discuss.

### CODE REVIEW

File: eras/conway/impl/cardano-ledger-conway.cabal [hunk 1]
```diff
@@ -36,7 +36,9 @@ library
         Cardano.Ledger.Conway.TxCert
 
     hs-source-dirs:   src
-    other-modules:
+    -- TODO: this is temporarily exposed so that we can write tests for this code in the cardano-ledger-test library
```
Comment (lehins) [inline eras/conway/impl/cardano-ledger-conway.cabal:39]: You can get all the instances from `Cardano.Ledger.Conway`. You should not need to expose these modules

File: libs/cardano-ledger-test/cardano-ledger-test.cabal [hunk 2]
```diff
@@ -81,12 +89,15 @@ library
         bytestring,
         cardano-data,
         cardano-crypto-class,
+        cardano-crypto-wrapper,
+        cardano-crypto,
```
Comment (lehins) [inline libs/cardano-ledger-test/cardano-ledger-test.cabal:93]: We don't do any Byron testing, so these packages should not be used

File: libs/cardano-ledger-test/cardano-ledger-test.cabal [hunk 3]
```diff
@@ -81,12 +89,15 @@ library
...
         cardano-ledger-allegra,
         cardano-ledger-alonzo:{cardano-ledger-alonzo, testlib},
         cardano-ledger-alonzo-test,
         cardano-ledger-api,
         cardano-ledger-babbage:{cardano-ledger-babbage, testlib},
         cardano-ledger-babbage-test,
+        cardano-ledger-byron,
```
Comment (lehins) [inline libs/cardano-ledger-test/cardano-ledger-test.cabal:100]: Same with this one

**Comment by MaximilianAlgehed** [comment]
@TimSheard to avoid trouble with merging when I'm on holiday, could we merge this into your branch now?

**Comment by lehins** [comment]
> @TimSheard to avoid trouble with merging when I'm on holiday, could we merge this into your branch now?

@MaximilianAlgehed It'll be better to do this as a separate PR into `master`. Tim's PR is already huge, we don't need to get it bigger.

We can merge it when you get back from vacation, no worries

**Comment by MaximilianAlgehed** [comment]
@lehins either way you want it. What I'm worried about is efforts diverging. This introduces some changes to Tim's stuff that I'd rather not have to rebase again as it can get quite hairy. However, I understand the aversion.

**Pull request closed without merging**